### PR TITLE
Add EIN field support

### PR DESCRIPTION
**Pull request by jjowdy**
EIN's are useful for business-related concerns (taxes, payroll, etc.). 

It'd be nice if there were a simple way to determine whether a TIN is actually an SSN versus an EIN, but the numeric spaces of the two overlap (both consist of nine digits).

If you have ideas for obvious refactorings to combine EIN and SSN into a common TIN class, I'm open to them, but I'm not sure it's worth it for such a simple class.

### CODE REVIEW

**Comment by eventualbuddha** [comment]
Thanks @jjowdy! @alexgorbatchev or @iamJoeTaylor, want to comment on this?

File: src/employer_identification_number_formatter.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import DelimitedTextFormatter from './delimited_text_formatter';
+
+/**
+ * @const
+ * @private
+ */
+var DIGITS_PATTERN = /^\d*$/;
```
Comment (iamJoeTaylor) [inline src/employer_identification_number_formatter.js]: Prefer `const` over var here.

PS I know this is from SSN but we still have some cleanup to do there, I'd rather add new stuff in the best ES6 style possible.

**Comment by iamJoeTaylor** [comment]
:+1: SGTM

We will need to follow up with wiki changes when this is merged.

File: src/employer_identification_number_formatter.js (hunk 1 above)
Comment (jjowdy) [inline src/employer_identification_number_formatter.js]: Fixed.

**Comment by iamJoeTaylor** [comment]
@eventualbuddha seem good to you?

**Comment by eventualbuddha** [comment]
:+1:

**Comment by iamJoeTaylor** [comment]
Small note on this, we are currently brainstorming how to break FieldKit apart into smaller modules so we don't balloon the code. Keep an eye out.

**Pull request merged**